### DISCUSSION TITLE
Remove Add Content when there can be added only 1 item to BlockList

### DISCUSSION THREAD

**Discussion by AdamczykMaciej** [umbraco/Umbraco-CMS#15395] [Features and ideas]
Remove Add Content when there can be added only 1 item to BlockList.

After adding an item to BlockList the button should disappear. Making it still visible suggests the user that there can be more items added.

**Comment by nul800sebastiaan** [comment]
I can't find the previous issue about this but the button is there to help people who are editing and realize that they wanted a different block then what they started with, but still have to copy some content between the old block and the new block. 

It would be really frustrating to have to delete the old block just so you can add a new one, you'd lose your previous work and would have to copy it outside Umbraco temporarily. Quite messy. So that's why the button is not removed!

**Reply by jdpnielsen** [reply]
A 'replace block with' functionality would be a better solution for this, in my opinion.

**Reply by nul800sebastiaan** [reply]
Not all block types are compatible for that of course.. so that also gets hairy really quickly! 😅

**Reply by jdpnielsen** [reply]
Not saying its simple, but it would be a much better solution.
I ended up creating a custom block list property editor just to be able to remove the button. Would be nice if it could at least be a setting.

**Reply by AdamczykMaciej** [reply]
We can leverage the built-in block-copying functionality. This would allow users to access the clipboard and review details of copied items by simply clicking on them. Upon clicking, the block would expand, displaying its properties and information.

**Comment by bjarnef** [comment]
I recall @nielslyngsoe mentioned this was by design in the newer UI like Media Picker 3, Block List etc.

Furthermore if there should only be one block and it always need to be added, one can consider to enable the single block mode, which scaffold the block and doesn't allow further blocks to be added.

**Reply by jdpnielsen** [reply]
Last i tested, the single block mode didn't change this behavior - but i might have to give it a test again.

**Reply by bjarnef** [reply]
The single block mode doesn't show the add button at all :)
As mentioned the block is scaffolded - either as inline block or the block opened in overlay.

**Reply by jdpnielsen** [reply]
Oh. Am i seeing a bug then? I'm on Umbraco v12.3.3
<img width="751" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/8746698/1e2b1dd6-e4a6-4cb6-940f-b3b2f8407c8d">
<img width="1027" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/8746698/7dbb3141-50ac-40e3-adcd-fcc767cc87a9">

**Reply by bjarnef** [reply]
Minimum and maximum also need to be set to 1 to enable single block mode.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/f0fdd8b2-cf4a-43ab-9bde-bda944b9e4ad)

**Reply by jdpnielsen** [reply]
Ah! Just tested it, and this works. I think it would make much more sense if it worked with minimum 0 also. This way, the block can be optional.
Additional it should also work with multiple types of blocks, so a user can select which block to use. The whole point for me is to support a polymorphic structure.
